backend/server/routers/programs.py

```diff
@@ -303,10 +325,43 @@ def get_structure_course_list(
         },
     },
 )
-def get_gen_eds(programCode: str):
-    """ fetches gen eds from file """
-    all_geneds = data_helpers.read_data("data/scrapers/genedPureRaw.json")[programCode]
-    return {"courses" : all_geneds}
+def get_gen_eds_route(programCode: str) -> Dict[str, Dict[str, str]]:
+    """ Fetches the geneds for a given program code """
+    course_list: List[str] = course_list_from_structure(get_structure(programCode, ignore="gened"))
+    return get_gen_eds(programCode, course_list)
+
+def get_gen_eds(
+        programCode: str, excluded_courses: Optional[List[str]] = None
+    ) -> Dict[str, Dict[str, str]]:
+    """
+    fetches gen eds from file and removes excluded courses
+    WARNING:
+        The `excluded_courses` is an optional parameter. This does
+        NOT mean you are free to ignore it. Default behaviour is to generate
+        a list of courses to exclude based on the `getStructure` call
+        for the respective programCode.
+        Be very careful leaving this field empty as `None` unless you are sure
+        you want to auto-generate excluded_courses. This can lead to unexpected
```

Comment:
This is confusing behaviour, why do we want to keep this the default?

Reply:
oop, this it outdated docs;
the new behaviour doesn't bootstrap `excluded_course` from `getStructure`, it just assumes that u want an empty list as ur default.

see:

```py
excluded_courses = excluded_courses if excluded_courses is not None else []
```

This was needed before because getStructure also builds geneds but, that's been changed now that getstructure is able to take an `ignore` param so i'll just delete these docs